internal/core/writer/spans.go

```diff
@@ -151,6 +153,8 @@ func (sw *SignalFxWriter) preprocessSpan(span *trace.Span) {
 		delete(span.Tags, dpmeta.NotHostSpecificMeta)
 	}
 
+	// adding smart agent version as a tag
+	span.Tags[constants.SmartAgentVersionTagKey] = os.Getenv(constants.AgentVersionEnvVar)
```

Comment:
There is a generated module in the constants package you are importing here called `version.go` that contains a string const `Version` that is much cleaner than going through this envvar.   That module isn't committed to git so that's probably why you didn't see it. The envvar with the version only exists to make it possible to pass it down to the collectd subprocess.